scripts/src/java/org/oppia/android/scripts/kdoc/KDocCheck.kt

```diff
@@ -0,0 +1,284 @@
+
+  // A list of all kotlin files in the repo to be analyzed.
+  val searchFiles = RepositoryFile.collectSearchFiles(
+    repoPath = repoPath,
+    expectedExtension = ".kt",
+    exemptionsList = kdocExemptionList
+  )
+
+  // A list of all the files missing KDoc(s).
+  val matchedFiles = searchFiles.filter { file ->
+    hasKdocFailure(file, kDocNotRequiredAnnotationEntryList)
+  }
+
+  if (matchedFiles.isNotEmpty()) {
+    throw Exception("KDOC CHECK FAILED")
+  } else {
+    println("KDOC CHECK PASSED")
+  }
+}
+
+/** IntelliJ Project object, which is required to generate a [KtFile] from a Kotlin file. */
+private val project by lazy {
+  val config = CompilerConfiguration()
+  config.put(CLIConfigurationKeys.MESSAGE_COLLECTOR_KEY, MessageCollector.NONE)
+  KotlinCoreEnvironment.createForProduction(
+    Disposer.newDisposable(),
+    config,
+    EnvironmentConfigFiles.JVM_CONFIG_FILES
+  ).project
+}
+
+/**
+ * Genrates a [KtFile] from a Kotlin file.
+ *
+ * @param codeString the string equivalent of the Kotlin file
+ * @param fileName name of the file
+ * @return the generated [KtFile] for the given file
+ */
+private fun createKtFile(codeString: String, fileName: String): KtFile {
+  return PsiManager.getInstance(project)
+    .findFile(LightVirtualFile(fileName, KotlinFileType.INSTANCE, codeString)) as KtFile
+}
+
+/**
+ * Checks whether a file has a KDoc presence failure.
+ *
+ * @param file the file to be checked
+ * @param kDocNotRequiredAnnotationEntryList the list of annotation entries which when present
+ *     on an element, the element does not needs to be checked for a KDoc.
+ * @return whether the file has a KDoc presence failure
+ */
+private fun hasKdocFailure(file: File, kDocNotRequiredAnnotationEntryList: List<String>): Boolean {
+  val ktFile = createKtFile(file.asString(), file.name)
+  return ktFile.children.fold(initial = false) { isFailing, elem ->
+    val childFailingKdoc = if (elem is KtElement) elementIterator(
+      elem,
+      file,
+      kDocNotRequiredAnnotationEntryList
+    ) else false
+    isFailing || childFailingKdoc
+  }
+}
+
+/**
+ * Returns the string equivalent of a file.
+ *
+ * @return the string equivalent
+ */
+private fun File.asString(): String {
+  val fileContents = StringBuilder(this.length().toInt())
+  Scanner(this).use { scanner ->
+    while (scanner.hasNextLine()) {
+      fileContents.append(scanner.nextLine() + System.lineSeparator())
+    }
+    return fileContents.toString()
+  }
+}
+
+/**
+ * Recursively iterates over an element and checks for a KDoc presence.
+ *
+ * @param elem the element over which we have to iterate
+ * @param file the file to be checked for the KDoc presence
+ * @param kDocNotRequiredAnnotationEntryList the list of annotation entries which when present
+ *     on an element, the element does not needs to be checked for a KDoc.
+ * @return whether the element is missing a KDoc
+ */
+private fun elementIterator(
```

Comment:
I'm not sure how to understand what this method is meant to do from the name alone. Could you rename it to be something more specific as to what it's actually doing? Generally, method names are verb phrases since they describe an action that's taking place.

Reply:
Done.